### ISSUE TITLE
Mask R-CNN only outputs bboxes but no masks on custom dataset

### ISSUE THREAD

**Issue by yeliudev** [open-mmlab/mmdetection#5005]
My custom dataset (15 classes) is in COCO format. After training the model, running the test produces the following results. Is it normal that for the same image, the number of output bboxes and masks from Mask R-CNN are not the same?

```
print(bbox, mask)

# Outputs
[array([[4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02],
       [1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32), array([[1.7340157e+01, 0.0000000e+00, 4.4142342e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.6833038e-01, 0.0000000e+00, 5.2999290e+01, 0.0000000e+00,
        6.2500000e-02],
       [4.7660495e+02, 0.0000000e+00, 5.0313907e+02, 0.0000000e+00,
        6.2500000e-02]], dtype=float32)] [[], [], [], [], [], [], [], [], [], [], [], [], [], [], []]
```

**Comment by yeliudev** [comment]
It seems that the bboxes are all (x1, 0, x2, 0). During training, the losses did not become `nan`. Is there any other possible reason that may cause the y-axis become 0?

**Comment by yeliudev** [comment]
I double-checked the rois, one the the rois is `tensor([  0.0000, 349.2560,   0.0000, 430.3175,   0.0000], device='cuda:0')`, which brings the undesired bboxes.

**Comment by yeliudev** [comment]
Hi @hhaAndroid! May I ask that is there a way to prevent rois with `area==0` in two-stage detectors in mmdet?

**Comment by yeliudev** [comment]
Fixed in #5011.
